spec/services/flow/steps/check_client_details_step_spec.rb

```diff
@@ -0,0 +1,29 @@
+require "rails_helper"
+
+RSpec.describe Flow::Steps::CheckClientDetailsStep do
+  let(:legal_aid_application) { create(:legal_aid_application) }
+
+  describe "#path" do
+    subject(:path) { described_class.path.call(legal_aid_application) }
+
+    it "returns the check_client_details_path when called" do
+      expect(path).to eq "/providers/applications/#{legal_aid_application.id}/check_client_details?locale=en"
+    end
+  end
+
+  describe "#forward" do
+    subject(:forward) { described_class.forward }
+
+    it "returns the about_financial_means step when called" do
+      expect(forward).to eq(:received_benefit_confirmations)
+    end
+  end
+
+  describe "#check_answers" do
+    subject(:check_answers) { described_class.check_answers }
+
+    it "returns nil" do
+      expect(check_answers).to be_nil
+    end
+  end
```

Comment:
Could use a generic custom matcher since we are going to do a lot of these

quick example:

```ruby
# spec/support/have_flow_step_args.rb

RSpec::Matchers.define :have_flow_step_args do |expected|
  results = { path: nil, forward: nil, check_answers: nil }

  match do |actual|
    results[:path] = actual.path.call(legal_aid_application) == expected[:path]
    results[:forward] = actual.forward == expected[:forward]
    results[:check_answers] = actual.check_answers == expected[:check_answers]

    results.values.all?(true)
  end

  failure_message do |actual|
    message = ""
    message += "expected #{actual.path.call(legal_aid_application)} to equal #{expected[:path]}\n" if actual.path.call(legal_aid_application) != expected[:path]
    message += "expected #{actual.forward} to equal #{expected[:forward]}\n" if actual.forward != expected[:forward]
    message += "expected #{actual.check_answers} to equal #{expected[:check_answers]}\n" if actual.check_answers != expected[:check_answers]
    message
  end

  description do
    "have matching flow step arguments"
  end
end
```

Reply:
I have added a commit to add a custom_matcher similar to the above. To make it generic, I have had to use a common name for the second paramater that is sometimes passed to the step lambda in the specs - I have called it args, but am not sure whether this is confusing?